File: akka-samples/akka-sample-main-scala/src/main/scala/sample/hello/Greeter.scala
/**
 *  Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com>
 */
package sample.hello

import akka.actor.Actor

object Greeter {
  case object Greet
  case object Done
}

class Greeter extends Actor {
  def receive = {
    case Greeter.Greet =>
      println("Hello World!")
      sender ! Greeter.Done
  }
}

File: akka-samples/akka-sample-main-java-lambda/src/main/java/sample/hello/Greeter.java
package sample.hello;

import akka.actor.AbstractActor;
import akka.japi.pf.ReceiveBuilder;

public class Greeter extends AbstractActor {

  public static enum Msg {
    GREET, DONE;
  }

  public Greeter() {
    receive(ReceiveBuilder.
      matchEquals(Msg.GREET, m -> {
        System.out.println("Hello World!");
        sender().tell(Msg.DONE, self());
      }).build());
  }
}
